R/bias.R

```diff
@@ -247,71 +246,65 @@ bias_range <- function(range, lower, upper,
+#' @seealso bias_quantile bias_sample
 #' @author Nikos Bosse \email{nikosbosse@@gmail.com}
 #' @examples
 #'
-#' predictions <- c(
-#'   705.500, 1127.000, 4006.250, 4341.500, 4709.000, 4821.996,
-#'   5340.500, 5451.000, 5703.500, 6087.014, 6329.500, 6341.000,
-#'   6352.500, 6594.986, 6978.500, 7231.000, 7341.500, 7860.004,
-#'   7973.000, 8340.500, 8675.750, 11555.000, 11976.500
+#' lower <- c(
+#'   6341.000, 6329.500, 6087.014, 5703.500,
+#'   5451.000, 5340.500, 4821.996, 4709.000,
+#'   4341.500, 4006.250, 1127.000, 705.500
 #' )
 #'
-#' quantiles <- c(0.01, 0.025, seq(0.05, 0.95, 0.05), 0.975, 0.99)
+#' upper <- c(
+#'   6341.000, 6352.500, 6594.986, 6978.500,
+#'   7231.000, 7341.500, 7860.004, 7973.000,
+#'   8340.500, 8675.750, 11555.000, 11976.500
+#' )
+#'
+#' range <- c(0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 95, 98)
 #'
 #' true_value <- 8062
 #'
-#' bias_quantile(predictions, quantiles, true_value = true_value)
+#' bias_range(
+#'   lower = lower, upper = upper,
+#'   range = range, true_value = true_value
+#' )
 #' @export
 #' @keywords metric
 
-bias_quantile <- function(predictions, quantiles, true_value) {
-  # check that predictions and quantiles have the same length
-  if (!length(predictions) == length(quantiles)) {
-    stop("predictions and quantiles must have the same length")
-  }
+bias_range <- function(lower, upper, range, true_value) {
 
-  if (anyNA(predictions)) {
-    quantiles <- quantiles[!is.na(predictions)]
-    predictions <- predictions[!is.na(predictions)]
-  }
-  # if there is no input, return NA
-  if (length(quantiles) == 0 || length(predictions) == 0) {
-    return(NA_real_)
+  if (!length(range) > 1 && !all(diff(range) > 0)) {
+    stop("Range must be increasing")
   }
 
-  if (0.5 %in% quantiles) {
-    median_prediction <- predictions[quantiles == 0.5]
-  } else {
-    # if median is not available, compute as mean of two innermost quantiles
-    median_prediction <-
-      0.5 * predictions[quantiles == max(quantiles[quantiles < 0.5])] +
-      0.5 * predictions[quantiles == min(quantiles[quantiles > 0.5])]
+  if (length(lower) != length(upper) || length(range) != length(lower)) {
+    stop("Inputs must have same length")
   }
 
-  if (true_value == median_prediction) {
-    bias <- 0
-    return(bias)
-  } else if (true_value < median_prediction) {
-    if (true_value < min(predictions)) {
-      bias <- 1
-    } else {
-      q <- max(quantiles[predictions <= true_value])
-      bias <- 1 - 2 * q
-    }
-  } else if (true_value > median_prediction) {
-    if (true_value > max(predictions)) {
-      bias <- -1
-    } else {
-      q <- min(quantiles[predictions >= true_value])
-      bias <- 1 - 2 * q
-    }
-  }
+  # Convert range to quantiles
+  quantiles <- c(
+    rev(abs(100 - range) / (2 * 100)),
+    abs(100 + range[range != 0]) / (2 * 100)
+  )
+
+  # Combine predictions
+  upper_without_median <- upper[range != 0]
+  predictions <- c(lower, upper_without_median)
```

Comment:
I *think* this should be 
```suggestion
  predictions <- c(rev(lower), upper_without_median)
```
as the lower values are decreasing.

Reply:
ok with this implemented the bias values in the vignette are the same as before. But probably should still add a test / check

Reply:
I added a suggestion to update the check that tests whether `bias_quantile()` and `bias_range()` give the same result. 
In addition I think testing explicitly whether values are sorted makes sense.